src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -31,20 +31,132 @@ from the [source tree] or using the Fuchsia SDK.
+1. Using the Fuchsia SDK and related `sysroot` to build and install a `rustc`
+   that can target Fuchsia.
+
+[rustup]: (https://rustup.rs/)
+[cargo]: (https://doc.rust-lang.org/cargo/)
+
+### Compiling a Rust binary targeting Fuchsia (Rustup/Cargo)
+
+The easiest way to build a Rust binary targeting Fuchsia is by allowing [rustup]
+to handle the installation of Fuchsia targets for you. This can be done by issuing
+the following commands:
+
+```sh
+rustup target add x86_64-fuchsia
+rustup target add aarch_64-fuchsia
+```
+
+After installing our Fuchsia targets, we can now compile a Rust binary that targets
+Fuchsia.
+
+To create our Rust project, we can issue a standard `cargo` command as follows:
+
+**From base working directory**
+```sh
+cargo new --name hello_fuchsia --bin hello_fuchsia_rust_crate
+```
+
+We can edit our `hello_fuchsia_rust_crate/src/main.rs` to include a test as follows:
+
+**`hello_fuchsia_rust_crate/src/main.rs`**
+```rust
+fn main() {
+    println!("Hello Fuchsia!");
+}
+
+#[test]
+fn it_works() {
+    assert_eq!(2 + 2, 4);
+}
+```
+
+In addition to the standard workspace created, we will want to create a
+`hello_fuchsia_rust_crate/build.rs` file to link necessary libraries during compilation:
+
+**`hello_fuchsia_rust_crate/build.rs`**
+```rust
+fn main() {
+  println!("cargo:rustc-link-search=native=<SDK_PATH>/arch/x64/sysroot/lib");
+  println!("cargo:rustc-link-search=native=<SDK_PATH>/arch/x64/lib");
+}
+```
+
+*Note: Make sure to fill out `<SDK_PATH>` with the path to the downloaded SDK.*
+
+In total, our new project will look like:
+
+**Current directory structure**
+```txt
+hello_fuchsia_rust_crate/
+┣━ src/
+┃  ┗━ main.rs
+┣━ cargo.toml
+┗━ build.rs
+```
+
+Finally, we can build our rust binary as:
+
+```sh
+cargo build --release --target x86_64-fuchsia --manifest-path hello_fuchsia_rust_crate/Cargo.toml
+```
+
+Now we have a Rust binary at `hello_fuchsia_rust_crate/target/x86_64-fuchsia/release/hello_fuchsia`,
+targeting our desired Fuchsia target.
+
+Copy this binary into an intermediate directory, to be used in later steps:
+
+```sh
+mkdir fuchsia_rust_binary
+cp hello_fuchsia_rust_crate/target/x86_64-fuchsia/release/hello_fuchsia fuchsia_rust_binary/hello_fuchsia
+```
+
+**Current directory structure**
+```txt
+hello_fuchsia_rust_crate/
+┣━ src/
+┃  ┗━ main.rs
+┣━ target/
+┃  ┣━ x86_64-fuchsia/
+┃  ┃  ┣━ release/
+┗━━┻━━┻━━┻━ hello_fuchsia       <-
+fuchsia_rust_binary/             | Copied binary
+┗━ hello_fuchsia                <-
+```
+
+### Compiling a Rust binary targeting Fuchsia (from source)
+
+An alternative to the first workflow is to compile Rust binaries
+targeting Fuchsia by using a `rustc` compiled from source.
 
 Before building Rust for Fuchsia, you'll need a clang toolchain that supports
 Fuchsia as well. A recent version (14+) of clang should be sufficient to compile
 Rust for Fuchsia.
 
-You'll also need a recent copy of the [Fuchsia SDK], which provides the tools
```

Comment:
This section is redundant but it may be worth keeping for the same reason as above. Alternatively, maybe we could make the prep section more obviously part of the process at the beginning.

Reply:
Good call -- I made the SDK the first part of the requirement now